mysys/mf_tempfile.c

```diff
@@ -110,6 +110,29 @@ File create_temp_file(char *to, const char *dir, const char *prefix,
     }
   }
 #elif defined(HAVE_MKSTEMP)
+#if O_TMPFILE
+  static int O_TMPFILE_works= 1;
+
+  if ((MyFlags & MY_TEMPORARY) && O_TMPFILE_works)
+  {
+    /* explictly don't use O_EXCL here has it has a different
+       meaning with O_TMPFILE
```

Comment:
I don't understand why you don't want to use O_EXCL here. But I presume it doesn't matter, we don't use linkat trick anyway.